daemon/containerd/image_builder.go

```diff
@@ -194,46 +195,47 @@ func newROLayerForImage(ctx context.Context, imgDesc *ocispec.Descriptor, i *Ima
 	if err != nil {
 		return nil, err
 	}
-	parent := identity.ChainID(diffIDs).String()
 
-	s := i.client.SnapshotService(i.snapshotter)
-	key := stringid.GenerateRandomID()
-	ctx, _, err = i.client.WithLease(ctx, leases.WithRandomID(), leases.WithExpiration(1*time.Hour))
-	if err != nil {
-		return nil, fmt.Errorf("failed to create lease for commit: %w", err)
-	}
-	mounts, err := s.View(ctx, key, parent)
-	if err != nil {
-		return nil, err
-	}
+	// TODO(vvoland): Check if image is unpacked, and unpack it if it's not.
+	imageSnapshotID := identity.ChainID(diffIDs).String()
 
-	tempMountLocation := os.TempDir()
-	root, err := os.MkdirTemp(tempMountLocation, "rootfs-mount")
+	snapshotter := i.StorageDriver()
+	_, lease, err := createLease(ctx, i.client.LeasesService())
 	if err != nil {
-		return nil, err
-	}
-
-	if err := mount.All(mounts, root); err != nil {
-		return nil, err
+		return nil, errdefs.System(fmt.Errorf("failed to lease image snapshot %s: %w", imageSnapshotID, err))
 	}
 
 	return &rolayer{
-		key:                key,
+		key:                imageSnapshotID,
 		c:                  i.client,
-		snapshotter:        i.snapshotter,
-		diffID:             digest.Digest(parent),
-		root:               root,
+		snapshotter:        snapshotter,
+		diffID:             "", // Image RO layer doesn't have a diff.
 		contentStoreDigest: "",
+		lease:              &lease,
 	}, nil
 }
 
+func createLease(ctx context.Context, lm leases.Manager) (context.Context, leases.Lease, error) {
+	lease, err := lm.Create(ctx,
+		leases.WithExpiration(time.Hour*24),
+		leases.WithLabels(map[string]string{
+			"org.mobyproject.lease.classicbuilder": "true",
+		}),
+	)
+	if err != nil {
+		return nil, leases.Lease{}, fmt.Errorf("failed to create a lease for snapshot: %w", err)
```

Comment:
Bit of a nit, but curious if there was a specific reason you didn't use a pointer return for the lease (to allow returning an explicit `nil` on failure)

Reply:
To save the caller from dereferencing the Lease pointer (which won't be nil anyway if the function returns a nil error). This function doesn't return a nil error with an empty lease, so there's no need to have the ability to encode an empty lease here.

Reply:
Oh! Right; I was focusing on this specific case, where we needed a pointer;

```go
		lease:              &lease,
```

But didn't look at the other cases. Gotcha; makes sense 😄